### PR TITLE
Make `get_infeasible_cost` return a cost value for each outcome.

### PR DESCRIPTION
**Pull request by saitcakmak**
Summary: Current implementation returns a single `M` value. This modifies it to return an `m`-dim tensor of infeasible cost values, each corresponding to one of the `m` outcomes.

Differential Revision: D35847505

### CODE REVIEW

File: botorch/acquisition/utils.py [hunk 1]
```diff
@@ -237,9 +237,13 @@ def objective(Y: Tensor, X: Optional[Tensor] = None):
             return Y.squeeze(-1)
 
     posterior = model.posterior(X, posterior_transform=posterior_transform)
-    lb = objective(posterior.mean - 6 * posterior.variance.clamp_min(0).sqrt()).min()
-    M = -(lb.clamp_max(0.0))
-    return M.item()
+    lb = objective(posterior.mean - 6 * posterior.variance.clamp_min(0).sqrt())
+    if lb.ndim < posterior.mean.ndim:
+        lb = lb.unsqueeze(-1)
+    # Take outcome-wise min. Looping in to handle batched models.
```
Comment (Balandat) [inline botorch/acquisition/utils.py:243]: So If I read this correctly, this computes the minimum across all batch dimensions? Is that what we want here? The previous implementation also did this so it's probably fine to get this in as is, but we may want to think about this some more if the batch dims correspond to different data and we want to extract them individually (this may be relevant for fully bayesian models - @dme65).

File: test/acquisition/test_objective.py [hunk 2]
```diff
@@ -320,7 +320,7 @@ def test_constrained_mc_objective(self):
                 obj=constrained_obj,
                 constraints=[feasible_con, infeasible_con],
                 samples=samples,
-                infeasible_cost=0.0,
+                infeasible_cost=torch.tensor([0.0], device=self.device, dtype=dtype),
```
Comment (Balandat) [inline test/acquisition/test_objective.py:323]: you're accepting `Union[Tensor, float]` - should we test support both input types?

**Review by Balandat — APPROVED** [review]

File: botorch/acquisition/utils.py (hunk 1 above)
Comment (saitcakmak) [reply botorch/acquisition/utils.py:243]: Yeah, that is an interesting thought. I guess it would depend on how different we expect each model in the batch to be.  Since they're all trained on the same set of inputs, I'd presume they'd be relatively similar, so one cost value should work for all. This would also affect the fantasized models, so maybe if we have two fantasy models, one on a large fantasy observation and the other on a small one, the cost value for one may be too conservative for the other? One could also argue that the 6 sigma we use here is too conservative to begin with. Anyways, I'm happy to revisit this later if we think it is worth exploring in more detail.

File: test/acquisition/test_objective.py (hunk 2 above)
Comment (saitcakmak) [reply test/acquisition/test_objective.py:323]: There are something like 6 tests in there. I only changed two of them to use tensors, so both float and tensor are being tested.
